apps/web-client/src/lib/db/cr-sqlite/customers.ts

```diff
@@ -30,11 +30,38 @@
 
 import type { DB, Customer, DBCustomerOrderLine, CustomerOrderLine, BookLine } from "./types";
 
+/**
+ * Retrieves all customers from the database.
+ * Returns basic customer information ordered by ID.
+ *
+ * @param {DB} db - Database connection
+ * @returns {Promise<Customer[]>} Array of customers with:
+ *   - id: Unique identifier
+ *   - fullname: Customer's full name (optional)
+ *   - email: Contact email (optional)
+ *   - updatedAt: Last modification timestamp
+ *   - deposit: Current deposit amount (optional)
+ */
 export async function getAllCustomers(db: DB): Promise<Customer[]> {
 	const result = await db.execO<Customer>("SELECT id, fullname, email, updatedAt, deposit FROM customer ORDER BY id ASC;");
 	return result;
 }
 
+/**
+ * Creates a new customer or updates an existing one.
+ * Uses customer ID as the unique identifier for upsert operations.
+ * Updates the customer's lastModified timestamp automatically.
+ * All fields except ID are optional and will only be updated if provided.
+ *
+ * @param {DB} db - Database connection
+ * @param {Customer} customer - Customer data including:
+ *   - id: Unique identifier (required)
+ *   - fullname: Customer's full name
+ *   - email: Contact email
+ *   - deposit: Current deposit amount
```

Comment:
I think we don't need to specify what's "inside" the types we're using.

```suggestion
 * @param {Customer} customer - Customer data
```